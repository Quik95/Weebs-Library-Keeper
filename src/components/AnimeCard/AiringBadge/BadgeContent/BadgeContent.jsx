import React, { memo } from "react";
import PropTypes from "prop-types";

//MUI
import Tooltip from "@material-ui/core/Tooltip";

//helpers
import formatDate from "../../../../helpers/formatDate";

const BadgeContent = memo(function BadgeContent({ timeUntilAiring }) {
  if (!timeUntilAiring) return null;

  const { relTime, absTime } = formatDate(timeUntilAiring);

  return (
    <Tooltip title={absTime} placement="top">
      <span>{relTime}</span>
    </Tooltip>
  );
});

BadgeContent.propTypes = {
  timeUntilAiring: PropTypes.number
};

export default BadgeContent;
