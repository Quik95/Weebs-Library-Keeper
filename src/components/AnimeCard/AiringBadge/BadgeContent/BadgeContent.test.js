import React from 'react'

import BadgeContent from './BadgeContent'

describe('Renders BadgeContent', () => {
  const wrapper = shallow(<BadgeContent absTime="15/12/2019 18:25" relTime="9h" />)

  it("Renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
