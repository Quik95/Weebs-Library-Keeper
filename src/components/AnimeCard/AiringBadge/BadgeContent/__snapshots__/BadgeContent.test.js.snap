// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders BadgeContent Renders correctly 1`] = `
<WithStyles(ForwardRef(Tooltip))
  placement="top"
  title="15/12/2019 18:25"
>
  <span>
    9h
  </span>
</WithStyles(ForwardRef(Tooltip))>
`;
